php/libraries/DefaultFilesUploadHandler.class.inc

```diff
@@ -0,0 +1,367 @@
+     *                                  does not exist, it will be created
+     *                                  automatically. If null, then the uploaded
+     *                                  files will be moved to $DEFAULT_MOVE_DIR.
+     * @param int       $dirGroup       if directory $moveDirPath has to be created
+     *                                  it will have this access group.
+     *
+     * @return nothing.
+     */
+    public function __construct(
+        LorisForm $lorisForm,
+        int $dirPermissions,
+        string $moveDirPath=null,
+        string $dirGroup=null
+    ) {
+        $this->_lorisForm      = $lorisForm;
+        $this->_moveDirPath    = is_null($moveDirPath)
+            ? self::$DEFAULT_MOVE_DIR : $moveDirPath;
+        $this->_dirPermissions = $dirPermissions;
+        $this->_dirGroup       = $dirGroup;
+
+        $this->_errors = array();
+    }
+
+    /**
+     * Adds an error (any kind) to the set of current errors.
+     *
+     * @param string $key      name of the LorisForm field associated to the
+     *                         uploaded file.
+     * @param string $errorMsg error message text.
+     *
+     * @return nothing.
+     */
+    protected function addError(string $key, string $errorMsg)
+    {
+        $this->_errors[$key] = $errorMsg;
+    }
+
+    /**
+     * Accessor for field $_errors. The keys are the names of the LorisForm
+     * fields used for file uploads. Only fields with errors are stored in this
+     * array.
+     *
+     * @return array array of all the errors found.
+     */
+    public function getErrors() : array
+    {
+        return $this->_errors;
+    }
+
+    /**
+     * Accessor for field $_lorisForm.
+     *
+     * @return LorisForm the LorisForm associated with this handler.
+     */
+    public function getLorisForm() : \LorisForm
+    {
+        return $this->_lorisForm;
+    }
+
+    /**
+     * Accessor for field $_dirPermissions.
+     *
+     * @return int value for field $_dirPermissions.
+     */
+    public function getDirPermissions() : int
+    {
+        return $this->_dirPermissions;
+    }
+
+    /**
+     * Accessor for field $_moveDirPath.
+     *
+     * @return string value for field $_moveDirPath.
+     */
+    public function getMoveDirPath() : string
+    {
+        return $this->_moveDirPath;
+    }
+
+    /**
+     * Accessor for field $_dirGroup.
+     *
+     * @return int value for field $_dirGroup.
+     */
+    public function getDirGroup() : string
+    {
+        return $this->_dirGroup;
+    }
+
+    /**
+     * Accessor for field $_movedFiles.
+     *
+     * @return array the final location of all files uploaded on the
+     *               server. Each key is the name of a LorisFirm field
+     *               used to upload a file and its corresponding value is the final
+     *               location for the uploaded file on the server.
+     */
```

Comment:
this needs to specify what needs to happen before the value returned is valid. Does process need to be called? Is it valid as soon as the constructor is called? Does something else need to populate it?

The public getter comment also disagrees with the comment on the member declaration of what the keys are and what the values are.